cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -144,17 +159,23 @@ std::optional<compute::Expression> ColumnChunkStatisticsAsExpression(
       return compute::or_(std::move(single_value), is_null(std::move(field_expr)));
     }
 
-    auto lower_bound =
-        compute::greater_equal(field_expr, compute::literal(std::move(min)));
-    auto upper_bound = compute::less_equal(field_expr, compute::literal(std::move(max)));
+    auto lower_bound = compute::greater_equal(field_expr, compute::literal(min));
```

Comment:
Perhaps the code could be better organized, to that NaN handling happens in a single place?
(note that `min->Equals(max)` can only be false if there's a NaN...)

Reply:
Apologies, I did not fully understand that. Can you elaborate on considering the inequality for min and max, and determining NaNs? Thanks!